### PR TITLE
Adds ACME - auto cert management

### LINKED ISSUE #5177: Add support for ACME
### Feature Request

I've been running a quickstart guide and it was quite tedious to get to a running teleport server. 
One of the obstacles running it in any environment is getting certs provisioned. 

Add support for ACME mode for proxy service. It does not have to be scalable on the first release.

### PR DESCRIPTION
**Pull request by klizhentas**
This commit fixes #5177

Initial implementation uses dir backend as a cache and is OK
for small clusters, but will be a problem for many proxies.

This implementation uses Go autocert that is quite limited
compared to Caddy's certmagic or lego.

Autocert has no OCSP stapling and no locking for cache for example.
However, it is much simpler and has no dependencies.
It will be easier to extend to use Teleport backend as a cert cache.

```yaml
proxy_service:
  public_addr: ['example.com']
  # ACME - automatic certificate management environment.
  #
  # It provisions certificates for domains and
  # valid subdomains in public_addr section.
  #
  # The sudomains are valid if there is a registered application.
  # For example, app.example.com will get a cert if app is a regsitered
  # application access app. The sudomain cookie.example.com is not.
  #
  # Teleport acme is using TLS-ALPN-01 challenge:
  #
  # https://letsencrypt.org/docs/challenge-types/#tls-alpn-01
  #
  acme:
    # By default acme is disabled.
    enabled: true
    # Use a custom URI, for example staging is
    #
    # https://acme-staging-v02.api.letsencrypt.org/directory
    #
    # Default is letsencrypt.org production URL:
    #
    # https://acme-v02.api.letsencrypt.org/directory
    uri: ''
    # Set email to receive alerts and other correspondence
    # from your certificate authority.
    email: 'alice@example.com'
```

### CODE REVIEW

**Comment by klizhentas** [comment]
Hey folks, I've been improving our docs from the perspective of an inexperienced user who has no idea about what is Teleport.

One of the obstacles for such user is the amount of work required to get a  single node cluster that is secure. It requires quite a lot of work with certificates, DNS records and other plumbing just to get started.

After I have added ACME support, Teleport became more fun for compared to the manual cert setup.

File: lib/web/app/match.go [hunk 1]
```diff
@@ -78,6 +81,60 @@ func MatchName(name string) Matcher {
...
 
-type appGetter interface {
+// Getter returns a list of registered apps
+type Getter interface {
+	// GetAppServers returns a list of app servers
 	GetAppServers(context.Context, string, ...services.MarshalOption) ([]services.Server, error)
 }
+
+// ResolveFQDN makes a best effort attempt to resolve FQDN to an application
+// running a root or leaf cluster.
```
Comment (a-palchikov) [inline lib/web/app/match.go:91]:
```suggestion
// running in a root or leaf cluster.
```

**Review by a-palchikov — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Comment by klizhentas** [comment]
@russjones @fspmarshall can you guys take a look at it when you have a chance in case if you any concerns.

**Review by fspmarshall — APPROVED** [review]
Code & UX looks good to me.  I'm not familiar enough with ACME to comment beyond that.

**Pull request merged**